tests/manage/monitoring/conftest.py

```diff
@@ -431,3 +442,421 @@ def corrupt_pg():
+    """
+    For the purpose of the workload storage utilization fixtures, get expected
+    pvc_size based on STORED and MAX AVAIL values (as reported by `ceph df`)
+    for given ceph pool and target utilization percentage.
+
+    This is only approximate, and it won't work eg. if each pool has different
+    configuration of replication.
+
+    Returns:
+        int: pvc_size for storage utilization job (in GiB, rounded)
+    """
+    # get STORED and MAX AVAIL of given ceph pool ...
+    ct_pod = pod.get_ceph_tools_pod()
+    ceph_df_dict = ct_pod.exec_ceph_cmd(ceph_cmd="ceph df")
+    ceph_pool = None
+    ceph_total_stored = 0
+    for pool in ceph_df_dict["pools"]:
+        ceph_total_stored += pool["stats"]["stored"]
+        if pool["name"] == ceph_pool_name:
+            ceph_pool = pool
+    if ceph_pool is None:
+        logger.error((
+            f"pool {ceph_pool_name} was not found "
+            f"in output of `ceph df`: {ceph_df_dict}"))
+    # If the following assert fail, the problem is either:
+    #  - name of the pool has changed (when this happens before GA, it's
+    #    likely ocs-ci bug, after the release it's a product bug),
+    #  - pool is missing (likely a product bug)
+    # either way, the fixture can't continue ...
+    assert ceph_pool is not None, f"pool {ceph_pool_name} should exist"
+    # ... to compute PVC size (values in bytes)
+    total = ceph_pool["stats"]["max_avail"] + ceph_total_stored
+    max_avail_gi = ceph_pool['stats']['max_avail'] / 2**30
+    logger.info(f"MAX AVAIL of {ceph_pool_name} is {max_avail_gi} Gi")
+    target = total * target_percentage
+    to_utilize = target - ceph_total_stored
+    pvc_size = round(to_utilize / 2**30)  # GiB
+    logger.info((
+        f"fixture is going to request {pvc_size} Gi volume "
+        f"to reach {target/2**30} Gi of total cluster utilization, which "
+        f"is {target_percentage*100}% of the total capacity"))
+    return pvc_size
+
+
+def fio_to_dict(fio_output):
+    """"
+    Parse fio output and provide parsed dict it as a result.
+    """
+    fio_output_lines = fio_output.splitlines()
+    for line_num, line in enumerate(fio_output_lines):
+        if line == "{":
+            break
+        else:
+            logger.info(line)
+    fio_parseable_output = "\n".join(fio_output_lines[line_num:])
+    fio_report = yaml.safe_load(fio_parseable_output)
+    return fio_report
+
+
+def workload_fio_storageutilization(
+        fixture_name,
+        target_percentage,
+        project,
+        fio_pvc_dict,
+        fio_job_dict,
+        fio_configmap_dict,
+        measurement_dir,
+        tmp_path,
+):
+    """
+    This function implements core functionality of fio storage utilization
+    workload fixture. This is necessary because we can't parametrize single
+    general fixture over multiple parameters (it would mess with test case id
+    and polarion test case tracking).
+    """
+    if fixture_name.endswith("rbd"):
```

Comment:
I think you can simply use CEPHFILESYSTEM and CEPHBLOCKPOOL instead of using the fixture name which is a hardcoded string that is not linked directly to the actual fixture name

Reply:
I'm checking the fixture name here to:
- avoid having to pass this information via another argument of `workload_fio_storageutilization`
- make sure that the name of the fixture have expected meaning

Reply:
That said, I could add storage_class_name and ceph_pool_name values into the constants module.

Reply:
The decision here is also not a big deal, so I'll be ok with either way